### DISCUSSION TITLE
Onverwacht gedrag bij opvoeren device locatie via de API

### DISCUSSION THREAD

**Discussion by CelineJansen** [kadaster-labs/sensrnet-home#174] [General]
Als ik een device op wil voeren en ik gebruik bijvoorbeeld het volgende object met RD coördinaten om de locatie te duiden:

"location": [
      161378.62114883581, 382970.16056510236, "9999"
    ]

geeft dat de volgende response:

  "message": [
    "location.location could not be interpreted as [long, lat, ..]"
  ]

long, lat gaat inderdaad wel goed:

    "location": [
      5.478934, 51.435799, "9999"
    ]

maw, RD x,y is geen geldige input, maar was niet juist de bedoeling om RD te nemen? Als beide opties wenselijk zijn, kan er onder locatie misschien een attribuut EPSG code komen. De gebruiker kan dan definiëren of de coördinaten conform WGS84 (epsg 4326) of RDnew + NAP (EPSG:7415) zijn?

**Comment by marcvanandel** [comment]
Goeie vraag (weer)! En niet zo gemakkelijk. We hebben hier al een paar issues over gemaakt over hoe dit zou moeten of wat het beste is:

- #93 Datamodel coordinaten -> laten we ESPG laten vallen en uitsluitend lat + lon (+ hoogte) gebruiken
- #143 verificatie bij NSGI -> goede, inhoudelijke discussie ... en staat nog open

Eerlijk gezegd weet ik niet goed wat standaard, gebruikelijk, het beste is 😬

**Reply by CelineJansen** [reply]
Het ligt er maar net een beetje aan wat je wil (in combinatie met wat technisch binnen jullie framework werkbaar is natuurlijk):

Voor mij als Geo persoon is RDnew + NAP (EPSG:7415) voor Nederland eigenlijk de meest logische optie. Jochem legt uit wat de voordelen zijn https://github.com/kadaster-labs/sensrnet-home/issues/143#issuecomment-782038313

WGS84 (EPSG:4326) past natuurlijk wel beter bij een internationaal karakter, dan is de vermelding [lat, lon, h] wel logischer dan [x, y, h]. ETRS89 zou dan een nauwkeuriger alternatief kunnen zijn, maar is wel onbekender voor de Geo-leek denk ik?

In de BGT en ook de SOR wordt het RD-NAP-coördinaatreferentiesysteem gehanteerd. In de viewer gebruiken jullie trouwens ook RD om de coördinaten van de cursor weer te geven? Dus dat maakt het onlogisch dat je vervolgens in WGS84 dingen moet opvoeren. 

Wat volgens mij ook van belang is nauwkeurigheid van de locatie. Nu is het geloof ik mogelijk om zelf een grote hoeveelheid decimalen te gebruiken. Daarmee wek je het idee dat de locatie heel nauwkeurig is. Gebruiken we WGS84, dan is dat niet perse het geval. Zeker niet omdat veel mensen in Nederland RD gebruiken, dat dan weer naar WGS84 gereproject moet worden (met als mogelijk gevolg extra onnauwkeurigheden).

In de SOR is het volgende voorstel gedaan:
"Coördinaten opgenomen bij een geometrie worden standaard uitgewisseld met een getalsnauwkeurigheid van 1 mm of het equivalent daarvan in graden. Voor RD en NAP komt dat overeen met de volgende nauwkeurigheden:

RD in meters 3 decimalen (1 mm);
NAP-hoogte in meters 3 decimalen (1 mm);
Alles wat nauwkeuriger is wordt afgerond op deze nauwkeurigheid van 3 decimalen. Afronding is volgens de volgende regel: 0.0015 -> 0.002; 0.0014 -> 0.001."

Let op: dit heeft dus toepassing op x,y,h en niet op lat,lon. Dan zijn andere standaarden misschien passender.
Je EPSG code kan je trouwens zien als een identifier voor je Coördinatensysteem (CRS).

We kunnen het hier misschien over hebben bij de FAB, ik ben geen echte expert zoals Jochem (mijn kennis beperkt zich tot een college van hem twee jaar geleden geloof ik ;) ) maar denk graag mee!

**Reply by marcvanandel** [reply]
Zeker goed om in het FAB te bespreken ... al denk ik wel dat dat enige voorbereiding vraagt. Laten we deze in ieder geval noemen zodat we de aandacht hier op vestigen, OK? (ik geef het door aan Roy 😉 )
